src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java

```diff
@@ -88,4 +94,32 @@ public static GoogleCredentials getApplicationDefaultCredentials() {
           ioEx);
     }
   }
+
+  /**
+   * Get an ID token from an Application Default Credential and Client Secrets. Note that the passed
+   * ADC must be properly scoped, passing improperly scoped credentials will result in a {@code
+   * SystemException}. Any other failure to obtain the token will result in an {@code IoException}.
+   */
+  public static IdToken getIdTokenFromApplicationDefaultCredentials(
+      GoogleCredentials applicationDefaultCredentials, GoogleClientSecrets clientSecrets)
```

Comment:
I moved the concept/object `clientSecrets` into a static final in `GoogleOauth` that gets initialized at static init time to guarantee that the JSON only ever gets read once.  This does therefore also guarantee that it gets read once even if `GoogleOauth.getClientSecrets()` is never called...

I was initially hoping to make it lazy-initialized on call to `getClientSecrets()`, but we call it from static functions and I didn't want to worry about race conditions on writing a static variable at runtime (even though I think this would only happen in a single-threaded context...).

I suspect that (1) `getClientSecrets()` is always called and (2) any further optimization is probably not worthwhile given this is reading out of the Jar and fairly lightweight.